source/compatibility.tex

```diff
@@ -2001,7 +2001,23 @@
 \tcode{<cstdalign>},
 \tcode{<cstdbool>}, and
 \tcode{<ctgmath>}.
-The \tcode{\#include} directives can simply be removed with no loss of meaning.
+To retain the same behavior:
+\begin{itemize}
+\item
+a \tcode{\#include} of \tcode{<ccomplex>} can be replaced by
+a \tcode{\#include} of \tcode{<complex>}\iref{complex.syn},
+\item
+a \tcode{\#include} of \tcode{<ctgmath>} can be replaced by
+a \tcode{\#include} of \tcode{<cmath>}\iref{cmath.syn} and
+a \tcode{\#include} of \tcode{<complex>},
+and
+\item
+a \tcode{\#include} of
+\tcode{<ciso646>},
+\tcode{<cstdalign>}, or
+\tcode{<cstdbool>}
+can simply be removed.
```

Comment:
Technically, any code using the `__*` macros is ill-formed (no diagnostic required) by [lex.name]p3.

Reply:
Errr... unless we specify a name, right? We do specify things like `__func__`, which are not UB to use.

Reply:
"some identifiers are reserved for use by C++ implementations and shall not be used otherwise; no diagnostic is required. [...] Each identifier that contains a double underscore `__` or begins with an underscore followed by an uppercase letter is reserved to the implementation for any use."

... sure suggests to me that using `__func__` is ill-formed, no diagnostic required. Sounds like we need a core issue.

Reply:
And what about `__cplusplus` and SD-6 feature test macros?

Reply:
@zygoloid: That cannot be the right reading. Not only do we have many macros with specified meaning, but also consider that C, which has similar wording, regularly introduces new keywords that use reserved names. It must be possible for the standard to specify names like that.